src/ec/packages/ec-component-timeline/timeline.html.twig

```diff
@@ -0,0 +1,120 @@
+{% spaceless %}
+
+{#
+Parameters:
+  - "toggleCollapsed": (string) (default: 'Show %d more items')
+  - "toggleExpanded": (string) (default: 'Show %d items')
+  - "hide" (array): (default: undefined)
+    - "from": (integer) (default: undefined) Positive integer
+    - "to": (integer) (default: undefined) Negative integer
+  - "items" (array) (default: []):
+    - "id": (string) (default: '')
+    - "label": (string) (default: '')
+    - "title": (string) (default: '')
+    - "content": (block) (default: '')
+  - "icon_path" (string) (default: ''): file containing the svg icons
+  - "extra_classes" (string) (default: '')
+  - "extra_attributes" (array) (default: []): format:
+    - "name" (string) (default: '')
+    - "value" (string) (default: '')
+#}
+
+{# Internal properties #}
+
+{% set _css_class = 'ecl-timeline2' %}
+{% set _extra_attributes = '' %}
+{% set _items = items|default([]) %}
+{% set _icon_path = icon_path|default('') %}
+{% set _toggle_collapsed = toggleCollapsed|default('Show %d more items') %}
+{% set _toggle_expanded = toggleExpanded|default('Hide %d items') %}
+{% set _hide_from = hide.from|default(items|length) %}
+{% set _hide_to = hide.to|default(items|length) %}
+
+{% set display_amount = items|length %}
+{% set hide_amount = 0 %}
+{% set bottom_amount = 0 %}
+
+{% if _hide_from > 0 %}
+  {% set bottom_amount = _hide_to|abs %}
+  {% set countbase = items|length - bottom_amount %}
+  {% set display_amount = _hide_from %}
+  {% set hide_amount = countbase - display_amount %}
+{% endif %}
+
+{# Internal logic - Process properties #}
+
+{% if extra_classes is defined and extra_classes is not empty %}
+  {% set _css_class = _css_class ~ ' ' ~ extra_classes %}
+{% endif %}
+
+{% if extra_attributes is defined and extra_attributes is not empty and extra_attributes is iterable %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %}
+
+{# Print the result #}
+
+<ol class="{{ _css_class }}"{{ _extra_attributes|raw }}
+    data-ecl-timeline="true">
+  {% if _items is not empty %}
+    {% for _item in _items %}
+      {# Display the show more button if required.#}
+      {% if hide_amount > 0 and loop.revindex == bottom_amount %}
+        <li class="ecl-timeline2__item ecl-timeline2__item--toggle">
+          {% include '../ec-component-button/button.html.twig' with {
+            label: toggleCollapsed|replace({'%d': countbase - display_amount}),
```

Comment:
The component should be as dumb as possible, meaning it should not replace the labels. The labels should be provided with the right text.